wordpress-mobile/WordPress-iOS
# Jetpack Social: Add social accounts screen in the pre-publishing flow

Refs #20783 

This shows the social accounts selector when the auto-sharing cell is tapped in the pre-publishing sheet. Tapping the Message row should similarly show you a text field screen that allows you to edit the sharing message. Navigating back from this screen should update the preview text within the message row.

In addition, through https://github.com/wordpress-mobile/WordPress-iOS/commit/807a40607e9a13a86a876510fe08f6257c3217fd I've modified the Core Data query to only include valid connections, to keep things simple.

Social accounts | Edit message
-|-
<img width="395" alt="Screenshot 2023-07-22 at 00 01 28" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/8be76a50-cde2-4d4e-8b35-ec3b7ac3a3f8"> |  <img width="395" alt="Screenshot 2023-07-22 at 00 03 53" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/8778aeef-3a27-4195-a2f8-db86caecbed6">

Note that most of the behaviors are not persisted yet in Core Data, and some components are also missing (the remaining shares text and the subscribe button). I've decided to add them in a separate PR to make this PR manageable.

## To test

- Launch the Jetpack app.
- Switch to a blog that has some connections.
- Create a new blog post, enter some title and text, and tap Publish.
- Tap the auto-sharing cell.
- 🔎 Verify that the connections are displayed.
- 🔎 Verify that the message row preview displays the post title.
- Tap on the message row.
- Make some changes to the text field.
- Tap Back.
- 🔎 Verify that the message preview text displays the new value.

### Row auto-disabling

When the blog has sharing limits, all the remaining social account rows (that are turned off) should be automatically disabled once the number of accounts that are turned on matches the remaining shares. To make testing easy, you can make some small code modifications in `PrepublishingSocialAccountsViewController`:

Replace line 47-51 with this test code:
```swift
self.connections = [
    .init(service: .facebook, account: "Foo", keyringID: 123, isOn: false),
    .init(service: .mastodon, account: "Bar", keyringID: 234, isOn: false)
]
```
and in line 53, modify so that the sharing limit has one limit remaining:
```swift
self.sharingLimit = .init(remaining: 1, limit: 30)
```

- Open the pre-publishing sheet by following the previous steps. The accounts should now be disabled by default.
- Tap one of the accounts to enable it.
- 🔎 Verify that the other account is now in a disabled state (non-interactive).
- Tap the account that's turned on.
- 🔎 Verify that the disabled state from the other account is removed.

### Accessibility

- Test with VoiceOver.
- Test with large accessibility sizes.

## Regression Notes
1. Potential unintended areas of impact
Should be none. Feature is unreleased.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the changes.

3. What automated tests I added (or what prevented me from doing so)
N/A.

PR submission checklist:


UI Changes testing checklist:
